### DISCUSSION TITLE
Intel Releases Intel TCB Recovery Guide for Q3 2023

### DISCUSSION THREAD

**Discussion by democ98** [gramineproject/gramine#1505] [General]
I'm using EPID attestation mode in Gramine,Developing applications on Linux with in-kernel sgx driver, according to this [message](https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/resources/q3-2023-intel-tcb-recovery-guidance.html#key-dates) from Intel,It seems that due to some vulnerabilities in Intel SGX, some programs need to be upgraded.
The link prompts to "update your Intel SGX Platform Software to at least v2.17 (for Linux* OS)", I don't quite understand, I use Gramine instead of Intel SGX PSW, For me,what action is required for this vulnerability?
@kailun-qin

**Comment by kailun-qin** [comment]
@democ98 Thanks for the question! Pls see my comments below.

> The link prompts to "update your Intel SGX Platform Software to at least v2.17 (for Linux* OS)", I don't quite understand, I use Gramine instead of Intel SGX PSW

Pls note that Gramine-SGX also requires Intel(R) SGX Platform Software (aka PSW) to be installed (e.g., Gramine remote attestation uses the AESM service of PSW), pls see our [env setup guide](https://gramine.readthedocs.io/en/stable/sgx-setup.html#set-up-the-host-environment) for the related info. So you need to update the PSW version in your env according to the TCB recovery guidance.

> For me,what action is required for this vulnerability?

Depending on your deployment model, besides the SGX SW update, you may also need microcode update (aka MCU) and BIOS update (pls refer to the TCB recovery guidance for details).

**Reply by democ98** [reply]
Thank you for your patience,Got it!
